### DISCUSSION TITLE
Fabric.js 原理与源码解析

### DISCUSSION THREAD

**Discussion by giscus** [keelii/fresh-blog#13] [Announcements]
# Fabric.js 原理与源码解析

芝兰生于深林，不以无人而不芳，君子修道立德，不为穷困而败节。

https://keelii.com/2021/05/09/fabricjs-internals

**Comment by Mariana-Yui** [comment]
刚开始是搜nestjs教程误打误撞了这个blog, 没想到最近需要学习fabricjs又点进来了, 大佬你多更点 爱看

**Reply by keelii** [reply]
fabricjs 已经要升级到 6.0 了，迁移到了 ts， 代码组件和类型继承关系都有了改善，学起来吧。

**Comment by zhangbowy** [comment]
我还在搞3.x版本

**Comment by enson0131** [comment]
收益匪浅，爱看，求更新

**Comment by moozeeli** [comment]
收益匪浅，感谢大佬的文章
